qcodes/instrument_drivers/tektronix/AWG70000A.py

```diff
@@ -0,0 +1,936 @@
+                           val_mapping={'direct': 'DIR',
+                                        'DCamplified': 'DCAM',
+                                        'AC': 'AC'}
+                           )
+
+        self.add_parameter('fgen_period',
+                           label='Channel {} {} period'.format(channel, fg),
+                           get_cmd='FGEN:CHANnel{}:PERiod?'.format(channel),
+                           unit='s',
+                           get_parser=float)
+
+        self.add_parameter('fgen_phase',
+                           label='Channel {} {} phase'.format(channel, fg),
+                           get_cmd='FGEN:CHANnel{}:PHASe?'.format(channel),
+                           set_cmd='FGEN:CHANnel{}:PHASe {{}}'.format(channel),
+                           unit='degrees',
+                           vals=vals.Numbers(-180, 180),
+                           get_parser=float)
+
+        self.add_parameter('fgen_symmetry',
+                           label='Channel {} {} symmetry'.format(channel, fg),
+                           set_cmd='FGEN:CHANnel{}:SYMMetry {{}}'.format(channel),
+                           get_cmd='FGEN:CHANnel{}:SYMMetry?'.format(channel),
+                           unit='%',
+                           vals=vals.Numbers(0, 100),
+                           get_parser=float)
+
+        self.add_parameter('fgen_type',
+                           label='Channel {} {} type'.format(channel, fg),
+                           set_cmd='FGEN:CHANnel{}:TYPE {{}}'.format(channel),
+                           get_cmd='FGEN:CHANnel{}:TYPE?'.format(channel),
+                           val_mapping={'SINE': 'SINE',
+                                        'SQUARE': 'SQU',
+                                        'TRIANGLE': 'TRI',
+                                        'NOISE': 'NOIS',
+                                        'DC': 'DC',
+                                        'GAUSSIAN': 'GAUSS',
+                                        'EXPONENTIALRISE': 'EXPR',
+                                        'EXPONENTIALDECAY': 'EXPD',
+                                        'NONE': 'NONE'})
+
+        ##################################################
+        # AWG PARAMETERS
+
+        # this command internally uses power in dBm
+        # the manual claims that this command only works in AC mode
+        # (OUTPut[n]:PATH is AC), but I've tested that it does what
+        # one would expect in DIR mode.
+        self.add_parameter(
+            'awg_amplitude',
+            label='Channel {} AWG peak-to-peak amplitude'.format(channel),
+            set_cmd='SOURCe{}:POWer {{}}'.format(channel),
+            get_cmd='SOURce{}:POWer?'.format(channel),
+            unit='V',
+            get_parser=lambda s: 2*10**((float(s)-10)/20),
+            set_parser=lambda V: 10 + 20 * np.log10(V/2),
+            vals=vals.Numbers(0.250, 0.500))
```

Comment:
How about using [SOURce[n]]:VOLTage[:LEVel][:IMMediate][:AMPLitude] as Carl suggested

Reply:
Yes, that seems more straightforward and less errorprone. I couldn't find that command in the manual, but I'll test is this afternoon.